### PR TITLE
Add controller to install components on master

### PR DESCRIPTION
**Pull request by csrwng**
Adds a controller to install components on a cluster that has accepted compute/infra nodes.

### CODE REVIEW

**Comment by csrwng** [comment]
@staebler @dgoodwin this is functional in that it will run the job, but the job is still failing due to the current playbooks checking for nodes with a particular label which assumes a static inventory. That's something that will need to be fixed on the openshift-ansible side.

File: pkg/apis/clusteroperator/types.go [hunk 1]
```diff
@@ -525,6 +539,18 @@ const (
 	// on this machine set.
 	MachineSetInstalled MachineSetConditionType = "Installed"
 
+	// MachineSetComponentsInstalling is true if OpenShift components are being installed on
+	// this machine set.
+	MachineSetComponentsInstalling MachineSetConditionType = "Installing"
```
Comment (staebler) [inline pkg/apis/clusteroperator/types.go]: I don't like that we have two different `MachineSetConditionType` constants that have the same value. Can we either (1) use the same const in both cases or (2) change the value for one of the consts. My preference would be (2) where, for example, "Installing" becomes "ComponentsInstalling" for `MachineSetComponentsInstalling`.

File: pkg/controller/components/components_controller.go [hunk 2]
```diff
@@ -0,0 +1,428 @@
+/*
+Copyright 2017 The Kubernetes Authors.
```
Comment (staebler) [inline pkg/controller/components/components_controller.go]: s/2107/2018/

File: pkg/controller/components/components_controller.go [hunk 3]
```diff
@@ -0,0 +1,428 @@
...
+	case controller.JobSyncProcessed:
+		return clusteroperator.MachineSetComponentsInstalled
+	case controller.JobSyncProcessingFailed:
+		return clusteroperator.MachineSetComponentsInstallationFailed
+	default:
+		return clusteroperator.MachineSetConditionType("")
+	}
+}
+
+func loggerForMachineSet(logger log.FieldLogger, machineSet *clusteroperator.MachineSet) log.FieldLogger {
```
Comment (staebler) [inline pkg/controller/components/components_controller.go]: Use `WithMachineSet` from pkg/logging/logging.go.

File: pkg/apis/clusteroperator/types.go (hunk 1 above)
Comment (csrwng) [reply pkg/apis/clusteroperator/types.go]: @staebler thx, that was not intentional, rather careless copy/paste. Will fix.

**Comment by csrwng** [comment]
comments addressed

**Pull request merged**